auth/auth-server/src/store/mod.rs

```diff
@@ -0,0 +1,79 @@
+//! Defines the bundle store and associated types
+use std::{collections::HashMap, sync::Arc};
+
+use auth_server_api::GasSponsorshipInfo;
+use renegade_circuit_types::wallet::Nullifier;
+use tokio::sync::RwLock;
+
+use crate::error::AuthServerError;
+
+pub mod helpers;
+
+/// Context of an external match bundle
+#[derive(Clone)]
+pub struct BundleContext {
+    /// The key description that settled the bundle
+    pub key_description: String,
+    /// The request ID of the bundle
+    pub request_id: String,
+    /// The SDK version that requested the bundle
+    pub sdk_version: String,
+    /// The gas sponsorship info for the bundle
+    pub gas_sponsorship_info: Option<GasSponsorshipInfo>,
+    /// Whether the bundle was sponsored
+    pub is_sponsored: bool,
+    /// The nullifier that was nullified as a result of the bundle being settled
+    pub nullifier: Nullifier,
+}
+
+struct StoreInner {
+    by_id: HashMap<String, BundleContext>,
+    by_null: HashMap<Nullifier, Vec<String>>,
+}
+
+impl StoreInner {
+    pub fn new() -> Self {
+        Self { by_id: HashMap::new(), by_null: HashMap::new() }
+    }
+}
+
+/// A thread-safe store for tracking bundle contexts by ID and nullifier.
+#[derive(Clone)]
+pub struct BundleStore {
+    inner: Arc<RwLock<StoreInner>>,
+}
+
+impl BundleStore {
+    pub fn new() -> Self {
+        Self { inner: Arc::new(RwLock::new(StoreInner::new())) }
+    }
+
+    pub async fn write(
+        &self,
+        bundle_id: String,
+        ctx: BundleContext,
+    ) -> Result<(), AuthServerError> {
+        let mut inner = self.inner.write().await;
```

Comment:
Curious what level of error handling is appropriate here, should I fully check for poisoned locks, missing key, pre-lock validations, etc. or is it sufficient to let the caller handle errors? My sense is that it's ok to let caller log errors since this is mainly used for metrics where errors aren't critical.